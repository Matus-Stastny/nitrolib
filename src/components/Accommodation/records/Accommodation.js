// @flow strict

import type { RoomType } from "../../../records/Room";

export type AccommodationType = {|
  hotelName: string,
  rating: number,
  location: {|
    latitude: number,
    longitude: number,
  |},
  address: string,
  photoUrl: string,
  rooms: Array<RoomType>,
  mapboxToken: string,
  zoom?: number,
  isMMB: boolean,
|};
