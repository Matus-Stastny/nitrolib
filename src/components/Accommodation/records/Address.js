// @flow strict

export type AddressType = ?{|
  +fullAddress: ?string,
|};
