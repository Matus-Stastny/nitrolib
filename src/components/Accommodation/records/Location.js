// @flow strict

export type LocationType = {|
  longitude: number,
  latitude: number,
  address: string,
  hotelName: string,
  mapboxToken: string,
  desktopWidth?: number,
  zoom?: number,
|};
