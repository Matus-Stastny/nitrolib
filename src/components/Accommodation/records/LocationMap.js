// @flow strict

export type LocationType = {|
  center: {|
    latitude: number,
    longitude: number,
  |},
  label: string,
  zoom: number,
  desktopWidth: number,
  mapboxToken: string,
|};
