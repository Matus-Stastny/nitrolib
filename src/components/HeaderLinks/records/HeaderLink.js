// @flow strict
export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};
