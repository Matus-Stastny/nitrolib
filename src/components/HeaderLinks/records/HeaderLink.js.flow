// @flow strict
export type ActiveTab = "travel" | null;

export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};

export type SearchForm = {|
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
|};
