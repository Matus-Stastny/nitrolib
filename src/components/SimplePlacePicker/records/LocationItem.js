// @flow strict
type Params = {|
  id: string,
  name: string,
  slug: string,
  code: string,
|};

type Gps = {|
  lat: number,
  lon: number,
|};

type Tag = {|
  month_from: number,
  month_to: number,
  tag: string,
|};

type City = {|
  autonomous_territory: boolean | null,
  code: string,
  continent: Params,
  country: Params,
|};

type Region = {|
  id: string,
  name: string,
  slug: string,
|};

type AlternativePoints = {|
  id: string,
  distance: number,
  duration: number,
|};

export type LocationType = "country" | "city" | "airport" | "continent" | "special";

export type LocationItem = {|
  active: boolean,
  alternative_departure_points: Array<AlternativePoints>,
  alternative_names: Array<string>,
  autonomous_territory: boolean | null,
  bus_stations?: number,
  city?: City,
  code: string,
  currency?: string,
  organizations?: Array<string>,
  neighbours?: Array<string>,
  dst_popularity_score: number,
  global_rank_dst: number,
  icao?: string,
  hotels: number,
  continent?: Params,
  country?: Params,
  population?: number,
  id: string,
  int_id?: string,
  location: Gps,
  name: string,
  rank: number,
  stations?: number,
  region: Region,
  slug?: string,
  code?: string,
  subdivision?: Params,
  airports?: number,
  special?: Array<mixed>,
  tags: Array<Tag>,
  timezone: string,
  type: LocationType,
|};
