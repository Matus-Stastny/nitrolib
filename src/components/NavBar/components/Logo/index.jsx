// @flow strict
import React from "react";
import styled, { css } from "styled-components";

import { logo } from "../../../../styles";
import type { ThemeProps } from "../../../../records/Theme";
import { themeDefault } from "../../../../records/Theme";
import { Consumer as BrandConsumer } from "../../../../services/brand/context";
import SvgLogo from "./SvgLogo";
import * as rtl from "../../../../styles/rtl";
import mq from "../../../../styles/mq";

const PoweredByKiwi = styled.span`
  color: ${({ theme }: ThemeProps) => theme.orbit.paletteInkDark};
  position: relative;
  font-size: ${({ theme }: ThemeProps) => theme.orbit.fontSizeTextSmall};
  margin-${rtl.left}: 10px;
  padding-${rtl.left}: 10px;
  font-weight: ${({ theme }: ThemeProps) => theme.orbit.fontWeightMedium};
  &:before {
    content: "";
    top: 0;
    bottom: 0;
    ${rtl.left}: 0;
    margin: auto;
    position: absolute;
    width: 2px;
    background: ${({ theme }: ThemeProps) => theme.orbit.paletteInkLighter};
    height: 30px;
  }

  ${mq.ltTablet(css`
    font-size: 10px;
    font-weight: ${({ theme }: ThemeProps) => theme.orbit.fontWeightNormal};;
    margin-${rtl.left}: 7px;
    padding-${rtl.left}: 7px;
  `)};
`;

PoweredByKiwi.defaultProps = {
  theme: themeDefault,
};

const LogoLinkStyled = styled.a`
  display: flex;
`;

const LogoStyled = styled.img`
  max-height: 35px;
  max-width: 140px;
  align-self: center;
  ${mq.ltTablet(css`
    display: none;
  `)};
`;

const LogoStyledMobile = styled.img`
  display: none;
  max-height: 40px;
  max-width: 40px;
  align-self: center;

  ${mq.ltTablet(css`
    display: block;
  `)};
`;

const Link = styled.a`
  display: flex;
`;

const logoBaseUrl = "https://images.kiwi.com/whitelabels";

const Logo = () => (
  <BrandConsumer>
    {({ id, home_redirect_url, name, powered_by_kiwi }) =>
      id === "kiwicom" ? (
        <Link href={home_redirect_url}>
          <SvgLogo height={logo.height} width={logo.width} title={name} />
        </Link>
      ) : (
        <>
          <LogoLinkStyled href={home_redirect_url}>
            <LogoStyled
              title={name}
              alt={name}
              srcSet={`${logoBaseUrl}/0x80/${id}.png?v=1 2x`}
              src={`${logoBaseUrl}/0x40/${id}.png?v=1`}
            />
            <LogoStyledMobile
              title={name}
              alt={name}
              srcSet={`${logoBaseUrl}/0x80/${id}-mobile.png?v=1 2x`}
              src={`${logoBaseUrl}/0x40/${id}-mobile.png?v=1`}
            />
          </LogoLinkStyled>
          {powered_by_kiwi && (
            <PoweredByKiwi>
              Powered by <br /> Kiwi.com
            </PoweredByKiwi>
          )}
        </>
      )
    }
  </BrandConsumer>
);

export default Logo;
