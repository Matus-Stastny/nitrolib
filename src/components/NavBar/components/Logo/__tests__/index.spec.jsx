// @flow
import * as React from "react";
import { shallow } from "enzyme";

import { brandDefault } from "records/Brand";

import Logo from "..";

describe("#Logo", () => {
  test("render", () => {
    const wrapper = shallow(<Logo height={50} width={75} brand={brandDefault} />);
    expect(wrapper).toMatchSnapshot();
  });
});
