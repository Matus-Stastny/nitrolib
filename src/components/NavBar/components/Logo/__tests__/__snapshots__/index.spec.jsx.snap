// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Logo render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
